heartbeat/osceip

```diff
@@ -0,0 +1,243 @@
+
+: ${OCF_RESKEY_osccli=${OCF_RESKEY_osccli_default}}
+: ${OCF_RESKEY_profile=${OCF_RESKEY_profile_default}}
+: ${OCF_RESKEY_region=${OCF_RESKEY_region_default}}
+
+meta_data() {
+    cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="osceip" version="1.0">
+<version>1.0</version>
+
+<longdesc lang="en">
+Resource Agent for Outscale Elastic IP Addresses.
+
+It manages Outscale Elastic IP Addresses with osc-cli.
+
+Credentials needs to be setup.
+
+See https://docs.outscale.com/en/userguide/Installing-and-Configuring-OSC-CLI.html for more information about osccli.
+</longdesc>
+<shortdesc lang="en">Outscale Elastic IP Address Resource Agent</shortdesc>
+
+<parameters>
+
+<parameter name="osccli" unique="0">
+<longdesc lang="en">
+command line tools for outscale services
+</longdesc>
+<shortdesc lang="en">OSC-CLI tools</shortdesc>
+<content type="string" default="${OCF_RESKEY_osccli_default}" />
+</parameter>
+
+<parameter name="profile">
+<longdesc lang="en">
+Valid OSC-CLI profile name (see ~/.osc/config and 'osc configure')
+</longdesc>
+<shortdesc lang="en">profile name</shortdesc>
+<content type="string" default="${OCF_RESKEY_profile_default}" />
+</parameter>
+
+<parameter name="elastic_ip" unique="1" required="1">
+<longdesc lang="en">
+reserved elastic ip for vm instance
+</longdesc>
+<shortdesc lang="en">reserved elastic ip for vm instance</shortdesc>
+<content type="string" default="" />
+</parameter>
+
+<parameter name="allocation_id" unique="1" required="1">
+<longdesc lang="en">
+reserved allocation id for vm instance
+</longdesc>
+<shortdesc lang="en">reserved allocation id for vm instance</shortdesc>
+<content type="string" default="" />
+</parameter>
+
+<parameter name="region" required="0">
+<longdesc lang="en">
+Region for osc resource (required for role-based authentication)
+</longdesc>
+<shortdesc lang="en">Region</shortdesc>
+<content type="string" default="${OCF_RESKEY_region_default}" />
+</parameter>
+</parameters>
+
+<actions>
+<action name="start"        timeout="30s" />
+<action name="stop"         timeout="30s" />
+<action name="monitor"      timeout="30s" interval="20s" depth="0" />
+<action name="migrate_to"   timeout="30s" />
+<action name="migrate_from" timeout="30s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate"     timeout="10s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+osceip_usage() {
+    cat <<END
+usage: $0 {start|stop|monitor|migrate_to|migrate_from|validate|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+osceip_start() {
+    osceip_monitor || return $OCF_SUCCESS
+
+    $oscCLI_CMD api LinkPublicIp --profile "${OCF_RESKEY_profile}" --PublicIp "${ELASTIC_IP}" --VmId "${INSTANCE_ID}" --AllowRelink True
+    RET=$?
+
+
+    if [ $RET -ne 0 ]; then
```

Comment:
You should add some more checks where osc-cli is run to return OCF_ERR_GENERIC or other error codes when osc-cli fails:
https://github.com/ClusterLabs/resource-agents/blob/main/doc/dev-guides/ra-dev-guide.asc#return-codes

Reply:
I add a check on elastic-ip parameter